graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasElasticsearchQuery.java

```diff
@@ -170,6 +181,15 @@ private Map<String, Object> runQueryWithLowLevelClient(String query) throws Atla
 
         } catch (IOException e) {
             LOG.error("Failed to execute direct query on ES {}", e.getMessage());
+
+            if (e instanceof SocketTimeoutException ||
+                    e instanceof UnknownHostException ||
+                    (e.getMessage() != null &&
+                            (e.getMessage().contains("Connection reset by peer") ||
+                                    e.getMessage().contains("Network error")))) {
+                throw new AtlasBaseException(AtlasErrorCode.SERVICE_UNAVAILABLE, e.getMessage());
+            }
```

Comment:
I see this part duplicated - can you write a separate method & move this logic & call the method wherever needed.